### PR TITLE
Force users to reference D2L.CodeStyle.Annotations

### PR DESCRIPTION
**Pull request by j3parker**
This is required so that GetTypeByMetadataName resolves for our
annotations. Without explicitly referencing annotations the semantic
model does not have complete information for those attributes.

### CODE REVIEW

File: src/D2L.CodeStyle.Analyzers/Build/MandatoryReferencesAnalyzer.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+					Diagnostic.Create(
+						Diagnostics.MustReferenceAnnotations,
+						Location.None
+					)
+				);
+			}
+		}
+
+		public static bool IsTheAnnotationsAssembly( MetadataReference mr ) {
+			return mr.Display.EndsWith( "\\D2L.CodeStyle.Annotations.dll" );
```
Comment (j3parker) [inline src/D2L.CodeStyle.Analyzers/Build/MandatoryReferencesAnalyzer.cs]: This isn't fantastic. We could maybe investigate strong-naming the assembly.

**Comment by j3parker** [comment]
After looking in to how Roslyn works more I think this is definitely what we should be doing. Kinda lame though.

**Comment by j3parker** [comment]
I'm not sure how to add a code fix for this because we'd have to locate D2L.CodeStyle.Annotations correctly.

I should update our VS templates to have these by default too.

File: src/D2L.CodeStyle.Analyzers/Build/MandatoryReferencesAnalyzer.cs (hunk 1 above)
Comment (JeffAshton) [reply src/D2L.CodeStyle.Analyzers/Build/MandatoryReferencesAnalyzer.cs]: If this is a path, Consider using Path.GetFileName then.

Comment (j3parker) [reply src/D2L.CodeStyle.Analyzers/Build/MandatoryReferencesAnalyzer.cs]: Hm, so the docs for `Display` say its a "path or name". I'm not really sure what that means but I've obviously assumed it's a path. I should probably figure out what's actually happening.

**Comment by j3parker** [comment]
Talked IRL

**Comment by j3parker** [comment]
The changes to the LMS necessary to turn this on are in PR.

**Pull request merged**